shared/actions/favorite.js

```diff
@@ -2,18 +2,84 @@
 
 import engine from '../engine'
 import * as Constants from '../constants/favorite'
-
+import {badgeApp} from './notifications'
+import {canonicalizeUsernames, parseFolderNameToUsers} from '../util/kbfs'
+import _ from 'lodash'
 import type {Folder, favorite_favoriteList_rpc} from '../constants/types/flow-types'
 import type {Dispatch} from '../constants/types/flux'
-
 import type {FavoriteList} from '../constants/favorite'
+import type {Props as FolderProps} from '../folders/render'
+
+const folderToProps = (folders: Array<Folder>, username: string = ''): FolderProps => { // eslint-disable-line space-infix-ops
+  let privateBadge = 0
+  let publicBadge = 0
+
+  const converted = folders.map(f => {
+    const users = canonicalizeUsernames(username, parseFolderNameToUsers(f.name))
+      .map(u => ({
+        username: u,
+        you: u === username,
+        broken: false
+      }))
+    const sortName = users.map(u => u.username).join(' ')
+
+    const groupAvatar = f.private ? (users.length > 2) : (users.length > 1)
+    const userAvatar = groupAvatar ? null : users[users.length - 1].username
+    const meta = null
+    // const meta = (__DEV__ && Math.random() < 0.2) ? 'new' : null // uncomment to test seeing new before we integrate fully
```

Comment:
Perhaps annotate this comment with "TODO" to match the other temporary comment below?